governance/program/src/tools/bpf_loader_upgradeable.rs

```diff
@@ -52,3 +56,43 @@ pub fn set_program_upgrade_authority<'a>(
         ],
     )
 }
+
+/// Asserts the program  is upgradable and its upgrade authority is a signer of the transaction
+pub fn assert_program_upgrade_authority_is_signer(
+    program_address: &Pubkey,
+    program_data_info: &AccountInfo,
+    program_upgrade_authority_info: &AccountInfo,
+) -> Result<(), ProgramError> {
+    if program_data_info.owner != &bpf_loader_upgradeable::id() {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+    let program_data_address = get_program_data_address(program_address);
+
+    if program_data_address != *program_data_info.key {
+        return Err(GovernanceError::InvalidProgramDataAccountAddress.into());
+    }
+
+    let upgrade_authority = match deserialize(&program_data_info.data.borrow())
+        .map_err(|_| GovernanceError::InvalidProgramDataAccountData)?
+    {
+        UpgradeableLoaderState::ProgramData {
+            slot: _,
+            upgrade_authority_address,
+        } => upgrade_authority_address,
+        _ => None,
+    };
+
+    match upgrade_authority {
```

Comment:
nit: using something like `ok_or` can make logic like this really slick, especially if it returns a result.  You can write
```
let upgrade_authority = upgrade_authority.ok_or(Err(GovernanceError::ProgramNotUpgradable.into()))?;
```
and then do `if upgrade_authority != ....`

Reply:
Done in d2c761d0